CMakeLists.txt

```diff
@@ -1,86 +1,66 @@
+# # Setup vcpkg - should be called before defining project()
+# run_vcpkg()
+
+# Set the project name and language
+project(myproject LANGUAGES CXX)
+
+# Initialize ProjectOptions
+# uncomment the options to enable them:
+ProjectOptions(
+      ENABLE_CACHE
+      # WARNINGS_AS_ERRORS
+      ENABLE_CPPCHECK
+      ENABLE_CLANG_TIDY
+      ENABLE_CONAN
+      ENABLE_COVERAGE
+      # ENABLE_IPO
+      # ENABLE_INCLUDE_WHAT_YOU_USE
+      # ENABLE_PCH
+      # PCH_HEADERS
+      # ENABLE_DOXYGEN
+      # ENABLE_USER_LINKER
+      # ENABLE_BUILD_WITH_TIME_TRACE
+      # ENABLE_UNITY
+      # ENABLE_SANITIZER_ADDRESS
+      # ENABLE_SANITIZER_LEAK
+      # ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+      # ENABLE_SANITIZER_THREAD
+      # ENABLE_SANITIZER_MEMORY
+      # CONAN_OPTIONS
+)
+# project_options is defined inside cmakelib
 target_compile_features(project_options INTERFACE cxx_std_17)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
-  option(ENABLE_BUILD_WITH_TIME_TRACE "Enable -ftime-trace to generate time tracing .json files on clang" OFF)
-  if(ENABLE_BUILD_WITH_TIME_TRACE)
-    target_compile_options(project_options INTERFACE -ftime-trace)
-  endif()
-endif()
-
-# Link this 'library' to use the warnings specified in CompilerWarnings.cmake
-add_library(project_warnings INTERFACE)
-
-# enable cache system
-include(cmake/Cache.cmake)
-
-# Add linker configuration
-include(cmake/Linker.cmake)
-configure_linker(project_options)
-
-# standard compiler warnings
-include(cmake/CompilerWarnings.cmake)
-set_project_warnings(project_warnings)
-
-# sanitizer options if supported by compiler
-include(cmake/Sanitizers.cmake)
-enable_sanitizers(project_options)
-
-# enable doxygen
-include(cmake/Doxygen.cmake)
-enable_doxygen()
-
-# allow for static analysis options
-include(cmake/StaticAnalyzers.cmake)
-
-option(BUILD_SHARED_LIBS "Enable compilation of shared libraries" OFF)
-option(ENABLE_TESTING "Enable Test Builds" ON)
-option(ENABLE_FUZZING "Enable Fuzzing Builds" OFF)
-
-# Very basic PCH example
-option(ENABLE_PCH "Enable Precompiled Headers" OFF)
-if(ENABLE_PCH)
-  # This sets a global PCH parameter, each project will build its own PCH, which is a good idea if any #define's change
-  #
-  # consider breaking this out per project as necessary
-  target_precompile_headers(
-    project_options
-    INTERFACE
-    <vector>
-    <string>
-    <map>
-    <utility>)
-endif()
-
-option(ENABLE_CONAN "Use Conan for dependency management" ON)
-if(ENABLE_CONAN)
-  include(cmake/Conan.cmake)
-  run_conan()
-endif()
+# Adding the src:
+add_subdirectory(src)
 
+# Adding the tests:
+option(ENABLE_TESTING "Enable the tests" ON)
 if(ENABLE_TESTING)
   enable_testing()
-  message("Building Tests. Be sure to check out test/constexpr_tests for constexpr testing")
+  message("Building Tests. Be sure to check out test/constexpr_tests for constexpr
+testing")
   add_subdirectory(test)
 endif()
 
+option(ENABLE_FUZZING "Enable the fuzz tests" OFF)
```

Comment:
This PR removes all of the CMake options from the project, except for these two: ENABLE_TESTING and ENABLE_FUZZING. I am not objecting to this; I am just curious to know why these options were not removed. Given the previous decisions made in this PR, I would have expected these options to disappear, and for the `enable_testing()` and `add_subdirectory()` calls to run by default.

Theoretically, given 2 projects derived from `cpp_starter_project`, where one project is dependent on the other project, will there be a name collision between the two `ENABLE_TESTING` options? I have not tried to do this myself, but I think it would be worthwhile to know the answer. I thought this was representative of the problem we were trying to avoid by removing all of the other options.

Reply:
The tests are a bit different than the GUI examples. Because tests are always written for all the projects, and constantly disabling/enabling them is not desirables. That's why this is a case where using an option is not a bad idea.
If CMake had a standardized variable for enabling and building the tests, then we did not have this issue here. In Rust, people will use `cargo test` to build and run the tests, and everything happens automatically. Because CMake lacks such command, we can treat `cmake -S ./ -B ./build -DENABLE_TESTING` like the build part of `cargo test`

Reply:
> The tests are a bit different than the GUI examples. Because tests are always written for all the projects, and constantly disabling/enabling them is not desirables. That's why this is a case where using an option is the better solution.

This sounds like an argument for restoring several of the options that were removed. Constantly enabling any of the sanitizer options cannot possibly be desirable.

Reply:
The global options are still available if you need them. Include `GlobalOptions.cmake`